### PR TITLE
docs/frontend-system: update remaining documentation section for 1.30 release

### PR DESCRIPTION
**Pull request by Rugvip**
This updates the rest of the documentation section to use the new patterns and APIs that are being shipped in the 1.30 release.

### CODE REVIEW

File: docs/frontend-system/building-plugins/02-testing.md [hunk 1]
```diff
@@ -93,98 +95,57 @@ A number of features (frontend extensions and overrides) are also accepted by th
...
-    );
-
-    expect(analyticsApiMock.getEvents()[0]).toMatchObject({
-      action: 'click',
-      subject: 'See details',
-    });
-  });
-});
-```
+When testing multiple extensions you may sometimes want to access the output of other extensions than the main test subject. You can use the `.query(ext)` method to query a different extension that has been added to the tester, either via its ID or passing the extension directly.
```
Comment (benjdlambert) [inline docs/frontend-system/building-plugins/02-testing.md:148]:
```suggestion
When testing multiple extensions you may sometimes want to access the output of other extensions than the main test subject. You can use the `.query(ext)` method to query a different extension that has been added to the tester, by passing the extension used with the `createExtensionTester(...).add(ext)`
```

Comment (benjdlambert) [reply docs/frontend-system/building-plugins/02-testing.md:148]: We dropped support for the string version to make the types nicer, happy to revisit if you think it's worth querying with just an ID, but wonder if it's worth it?

Comment (Rugvip) [reply docs/frontend-system/building-plugins/02-testing.md:148]: ah yep no that's good 👍

File: docs/frontend-system/building-apps/01-index.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ It is possible to enable, disable and configure extensions individually in the `
 
 ### Customize or override built-in extensions
 
-Previously you would customize the application route, components, apis, sidebar, etc. through the code in `App.tsx`. Now we want you to write less code and install more extensions to customize your Backstage instance. See [here](../building-plugins/03-extension-types.md) which types of extensions are available for you to customize your application.
+Previously you would customize the application route, components, apis, sidebar, etc. through the code in `App.tsx`. Now we want you to write less code and install more extensions to customize your Backstage instance. See the [extension blueprints](../building-plugins/03-common-extension-blueprints.md) section for a list of common extension kinds that are available for you to customize and extend your application.
```
Comment (freben) [inline docs/frontend-system/building-apps/01-index.md:81]:
```suggestion
Previously you would customize the application routes, components, apis, sidebar, etc. through the code in `App.tsx`. Now we want to allow the same thing to be achieved while writing less code and instead installing more extensions to customize your Backstage instance. See the [extension blueprints](../building-plugins/03-common-extension-blueprints.md) section for a list of common extension kinds that are available for you to customize and extend your application.
```

File: docs/frontend-system/building-plugins/01-index.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ The plugin ID should be a lowercase dash-separated string, while the plugin inst
...
 
-To create a new extension you typically use pre-defined [extension creators](../architecture/20-extensions.md#extension-creators), provided either by the framework itself or by other plugins. In this case we'll need to use `createPageExtension` and `createNavItemExtension`, both from `@backstage/frontend-plugin-api`. We will also need to [create a route reference](../architecture/36-routes.md#creating-a-route-reference) to use as a reference for out page, allowing us to dynamically create URLs that link to our page.
+To create a new extension you typically use pre-defined [extension blueprints](../architecture/23-extension-blueprints.md), provided either by the framework itself or by other plugins. In this case we'll use `PageBlueprint` and `NavItemBlueprint`, both from `@backstage/frontend-plugin-api`. We will also need to [create a route reference](../architecture/36-routes.md#creating-a-route-reference) to use as a reference for out page, allowing us to dynamically create URLs that link to our page.
```
Comment (freben) [inline docs/frontend-system/building-plugins/01-index.md:52]:
```suggestion
To create a new extension you typically use pre-defined [extension blueprints](../architecture/23-extension-blueprints.md), provided either by the framework itself or by other plugins. In this case we'll use `PageBlueprint` and `NavItemBlueprint`, both from `@backstage/frontend-plugin-api`. We will also need to [create a route reference](../architecture/36-routes.md#creating-a-route-reference) to use as a reference for our page, allowing us to dynamically create URLs that link to our page.
```

**Review by freben — APPROVED** [review]

**Pull request merged**